core/src/smartcontracts/isi/asset.rs

```diff
@@ -335,7 +334,89 @@ pub mod isi {
                     .ok_or(MathError::Overflow)?;
                 wsv.metrics.tx_amounts.observe(f64::from(*quantity));
 
-                Ok(AssetEvent::Added(destination_asset_id.clone()))
+                Ok(AssetEvent::Added(self.destination_id.clone()))
+            })
+        }
+    }
+
+    impl<W: WorldTrait> Execute<W> for Transfer<Asset, u128, Asset> {
+        type Error = Error;
+
+        #[metrics(+"transfer_qty_asset")]
+        fn execute(
+            self,
+            _authority: <Account as Identifiable>::Id,
+            wsv: &WorldStateView<W>,
+        ) -> Result<(), Self::Error> {
+            assert_matching_definitions(
+                &self.source_id,
+                &self.destination_id,
+                wsv,
+                AssetValueType::BigQuantity,
+            )?;
+
+            wsv.asset_or_insert(&self.destination_id, 0_u128)?;
+            wsv.modify_asset(&self.source_id, |asset| {
+                let quantity: &mut u128 = asset
+                    .try_as_mut()
+                    .map_err(eyre::Error::from)
+                    .map_err(|e| Error::Conversion(e.to_string()))?;
+                *quantity = quantity
+                    .checked_sub(self.object)
+                    .ok_or(Error::Math(MathError::NotEnoughQuantity))?;
+
+                Ok(AssetEvent::Removed(self.source_id.clone()))
+            })?;
+            wsv.modify_asset(&self.destination_id, |asset| {
+                let quantity: &mut u128 = asset
+                    .try_as_mut()
+                    .map_err(eyre::Error::from)
+                    .map_err(|e| Error::Conversion(e.to_string()))?;
+                *quantity = quantity
+                    .checked_add(self.object)
+                    .ok_or(MathError::Overflow)?;
+                // wsv.metrics.tx_amounts.observe(f64::from(*quantity));
```

Comment:
Commented code

Reply:
I'm still thinking on the best way to do it. `u128` can be represented by `f64`, since `f64` uses scientific notation, has an 11 bit exponent and we don't care about precision. I don't want to hand-roll an implementation and see if there's a crate that has this conversion. If not, just right-shift by 64 bits, turn `u128 → u64` convert to `f64` add `64` to the exponent.

Reply:
https://docs.rs/floatconv/0.2.8/floatconv/fast/index.html

Reply:
Nevermind. We don't need the extra precision so `*quantity as f64` works just as well.